internal/mikrotik/provider.go

```diff
@@ -170,6 +170,13 @@ func cleanupChanges(changes *plan.Changes) *plan.Changes {
 
 	duplicates := []*endpoint.Endpoint{}
 
+	for _, create := range changes.Create {
+		if !create.RecordTTL.IsConfigured() {
+			create.RecordTTL = endpoint.TTL(p.client.TTL)
+			newChanges.Create = append(newChanges.Create, create)
+		}
+	}
+
```

Comment:
I think I found a new issue here. The check and enforcement of the default ttl is currently only done on record creation. This means that external DNS will create the record with the default ttl correctly, but then at the next loop it will update it back to 0. We should probably do a similar check for updating records

Reply:
Also, when we are initializing `newChanges` a few lines up, we're setting `newChanges.Create = changes.Create`. This causes the entire list to be duplicated. We need to set:

```go
	newChanges := &plan.Changes{
		Create:    []*endpoint.Endpoint{},
		Delete:    changes.Delete,
		UpdateOld: []*endpoint.Endpoint{},
		UpdateNew: []*endpoint.Endpoint{},
	}
```